src/optimizer/query_rewrite.c

```diff
@@ -3499,6 +3500,651 @@ qo_reduce_outer_joined_tables (PARSER_CONTEXT * parser, PT_NODE * spec, PT_NODE
+ *           -> do not change.
+ */
+static void
+qo_reduce_joined_referenced_tables (PARSER_CONTEXT * parser, PT_NODE * query)
+{
+  PT_NODE *curr_pk_spec = NULL, *prev_pk_spec = NULL, *next_pk_spec = NULL;
+  PT_NODE *curr_fk_spec = NULL;
+  SPEC_CNT_INFO info = { NULL, 0, 0, NULL };
+  PT_NODE *backup_where = NULL;
+  PT_NODE *curr_pred = NULL, *prev_pred = NULL, *next_pred = NULL, *parent_pred = NULL;
+  PT_NODE *exclude_spec_point_list = NULL, *exclude_spec_point = NULL, *exclude_spec = NULL;
+  PT_NODE *pred_point_list = NULL, *pred_point = NULL;
+  PT_NODE *curr_pred_point = NULL, *prev_pred_point = NULL, *next_pred_point = NULL;
+  PT_NODE *append_pred_list = NULL;
+  PT_NODE *curr_append_pred = NULL, *prev_append_pred = NULL, *next_append_pred = NULL;
+  int i;
+
+  if (parser == NULL || query == NULL || query->node_type != PT_SELECT)
+    {
+      return;
+    }
+
+  if (query->info.query.q.select.hint & PT_HINT_NO_ELIMINATE_JOIN)
+    {
+      return;
+    }
+
+  if (query->info.query.q.select.where != NULL && query->info.query.q.select.where->or_next != NULL)
+    {
+      return;
+    }
+
+  while (true)
+    {
+      bool has_reduce = false;
+
+      for (prev_pk_spec = NULL, curr_pk_spec = query->info.query.q.select.from; curr_pk_spec != NULL;
+	   prev_pk_spec = curr_pk_spec, curr_pk_spec = curr_pk_spec->next)
+	{
+	  MOP curr_pk_mop = NULL;
+	  SM_CLASS_CONSTRAINT *curr_pk_cons = NULL;
+	  bool is_skip_pk_spec = false;
+	  bool is_reducible = false;
+
+	  assert (PT_NODE_IS_SPEC (curr_pk_spec));
+
+	  for (exclude_spec_point = exclude_spec_point_list; exclude_spec_point != NULL;
+	       exclude_spec_point = exclude_spec_point->next)
+	    {
+	      exclude_spec = exclude_spec_point;
+	      CAST_POINTER_TO_NODE (exclude_spec);
+
+	      if (exclude_spec == curr_pk_spec)
+		{
+		  is_skip_pk_spec = true;
+		  break;
+		}
+	    }
+
+	  if (is_skip_pk_spec)
+	    {
+	      continue;		/* curr_pk_spec->next */
+	    }
+
+	  /* PT_ALL is not supported. */
+	  if (curr_pk_spec->info.spec.only_all == PT_ALL)
+	    {
+	      continue;		/* curr_pk_spec->next */
+	    }
+
+	  /* CTEs and derived tables are excluded. */
+	  if (PT_SPEC_ENTITY_NAME (curr_pk_spec) == NULL || !PT_NODE_IS_NAME (PT_SPEC_ENTITY_NAME (curr_pk_spec)))
+	    {
+	      continue;		/* curr_pk_spec->next */
+	    }
+
+	  switch (curr_pk_spec->info.spec.join_type)
+	    {
+	    case PT_JOIN_NONE:
+	    case PT_JOIN_INNER:
+	    case PT_JOIN_NATURAL:
+	      break;
+
+	    case PT_JOIN_CROSS:
+	    case PT_JOIN_LEFT_OUTER:
+	    case PT_JOIN_RIGHT_OUTER:
+	    case PT_JOIN_FULL_OUTER:
+	    case PT_JOIN_UNION:
+	      continue;		/* curr_pk_spec->next */
+	    }
+
+	  PT_SPEC_GET_DB_OBJECT (curr_pk_spec, curr_pk_mop);
+	  if (curr_pk_mop == NULL)
+	    {
+	      ASSERT_ERROR ();
+	      goto cleanup;
+	    }
+
+#if defined(CUBRID_DEBUG) || defined(CUBRID_DEBUG_TEST)
+	  printf ("[debug] [parent] (1/7) table: %s\n", pt_print_alias (parser, curr_pk_spec));
```

Comment:
Instead of printf(), `er_log_debug(ARG_FILE_LINE,
		    "[parent] (1/7) table: %s\n", pt_print_alias (parser, curr_pk_spec));`

Reply:
Debug message output became unnecessary, so i removed it.

Reply:
Added er_log_debug() again. Thank you.